### PR TITLE
Rename I2C Drivers Folder

### PR DESCRIPTION
**Pull request by shaggygi**
This is part of #126

Renames I2C Drivers folder to Devices.

### CODE REVIEW

**Comment by shaggygi** [comment]
@krwq This first change doesn't appear to break anything (yet).  Do you want me to add more commits under this PR for the namespace updates or create another PR specifically for those?

**Comment by krwq** [comment]
@shaggygi let's do it separately since we will likely squash and merge which can break the diff

**Comment by shaggygi** [comment]
@krwq I just pushed the button as I saw your response

**Comment by joperezr** [comment]
> I just pushed the button as I saw your response

That is fine, we will take it as is, I'm just taking a final look and making sure CI is green.

**Comment by joperezr** [comment]
The reason why your PR is red is because you premptively changed samples on bindings that haven't ingested the changes on this PR. Remember that the samples use a package reference to System.Device.Gpio, so it will take us first to merge this PR, and then to ingest those changes in order to be able to use the new namespaces in the samples. For this PR, we probably only want the namespace/file renames plus anything that would break the build, and then in the ingestion PR we will handle the namespace renames on the samples.

**Comment by shaggygi** [comment]
Gotcha.  So I basically need to revert back to b86f15d.

**Comment by krwq** [comment]
so basically:
- do renames only in System.Device.Gpio (no renames in the samples)
- when new package gets produced we will get PR which becomes red and the renames for samples should be added to that PR

**Comment by shaggygi** [comment]
Well snap.  DeviceApiTester is referencing directly to core project unlike bindings. So that's why it failed last time.  Let's try again 😦

File: src/System.Device.Gpio/System/Device/I2c/Devices/I2cDevice.cs [hunk 1]
```diff
@@ -2,7 +2,7 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-namespace System.Device.I2c
+namespace System.Device.I2c.Devices
```
Comment (krwq) [inline src/System.Device.Gpio/System/Device/I2c/Devices/I2cDevice.cs]: Is this change expected as well? I'd expect I2cDevice to live under System.Device.I2c and implementations under .Devices. cc: @joperezr

Comment (shaggygi) [reply src/System.Device.Gpio/System/Device/I2c/Devices/I2cDevice.cs]: I was thinking this was part of...

> 3. I2cDevice.cs moved to Devices folder.

Comment (krwq) [reply src/System.Device.Gpio/System/Device/I2c/Devices/I2cDevice.cs]: don't have strong opinion, it just feels weird from naming point of view. @joperezr, thoughts?

Comment (joperezr) [reply src/System.Device.Gpio/System/Device/I2c/Devices/I2cDevice.cs]: mm I also thought about this, but I think it does make sense to have it moved to this new namespace as well. It makes sense that a Device is on the Devices namespace :)

Comment (krwq) [reply src/System.Device.Gpio/System/Device/I2c/Devices/I2cDevice.cs]: @joperezr I'm wondering, perhaps we should make Windows/LinuxI2cDevice internal (I don't see good justification for having them right now) and just don't rename namespace (just add directories)

Comment (krwq) [reply src/System.Device.Gpio/System/Device/I2c/Devices/I2cDevice.cs]: We talked offline with @joperezr and we will only have 1 I2cDevice class which collapses the logic of both Linux and Windows and I2cDevice.Create will become `new I2cDevice`. We are also thinking of doing similar thing for board drivers but that one has a bit more justification on having them separate so we need to figure out what should happen.

Comment (joperezr) [reply src/System.Device.Gpio/System/Device/I2c/Devices/I2cDevice.cs]: We will do that in a separate PR though, so we can take this in first and we will make the required changes subsequently

Comment (shaggygi) [reply src/System.Device.Gpio/System/Device/I2c/Devices/I2cDevice.cs]: Do I need to create the next PR to update the bindings to reflect these changes or wait?  Also, I was going to create another PR similar to this for SPI.  Do I need to hold off on that base on the planned changes?

**Pull request merged**